float32 x
float32 y
float32 z
float32 yaw
float32 cost
float32 battery
float32 reward